ion/readlocalsymboltable.go

```diff
@@ -0,0 +1,206 @@
+package ion
+
+import "fmt"
+
+// ReadLocalSymbolTable reads and installs a new local symbol table.
+func readLocalSymbolTable(r Reader, cat Catalog) (SymbolTable, error) {
+	if r.IsNull() {
+		r.Clear()
+		return V1SystemSymbolTable, nil
+	}
+
+	if err := r.StepIn(); err != nil {
+		return nil, err
+	}
+
+	var imps []SharedSymbolTable
+	var syms []string
+
+	foundImport := false
+	foundLocals := false
+
+	for r.Next() {
+		var err error
+		switch *r.FieldName() {
```

Comment:
What if r.FieldName() is nil? Let's add an error case for that.

Reply:
Added a check for nil fieldname